horovod/run/util/cache.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2019 Uber Technologies, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import os
+import datetime
+import cloudpickle
+import errno
+import threading
+
+
+class Cache(object):
+    """
+    Cache the function calls
+    """
+
+    def __init__(self, cache_folder, cache_staleness_threshold_in_minutes,
+                 parameters_hash):
+
+        self._cache_file = os.path.join(cache_folder, 'cache.txt')
```

Comment:
What format is this file?  If it's json/yaml, let's call it appropriately.  If it's binary, could suffix with `.bin`, and remember to open() with 'b'